src/const_eval.md

```diff
@@ -40,6 +40,7 @@ to be ran.
 * [Cast] expressions, except pointer to address and
   function pointer to address casts.
 * Calls of const functions and const methods
+* [let statement]s and thus irrefutable [patterns]
```

Comment:
We should mention the restriction wrt. `&&` here. (also end sentence with punctuation for consistency, and the bullet above...).

Reply:
Also is this in the right place? `let` is not an expression...